Various Mechlab/Tournament UI Fixes

Nopey opened the pull request:
Fixes #1106

katajakasa reviewed (approved):
Tested, works, code seems fine. LGTM.

Vagabond commented on src/game/scenes/mechlab/lab_menu_main.c:
```diff
@@ -252,7 +266,7 @@ static const spritebutton_tick_cb tick_cbs[] = {
     lab_menu_tick_in_tournament, // lab_menu_tick_delete,
     lab_menu_tick_in_tournament, // lab_menu_tick_sim,
     NULL,                        // lab_menu_tick_quit,
-    lab_menu_tick_in_tournament, // lab_menu_tick_tournament,
```
This should be disabled when there's not a player loaded, at least.

Nopey replied:
The loop in lab_menu_main_create handles it.

that loop now also now sets supports_select, so the buttons the loop disables can't be selected.

Vagabond commented:
Erasing the "last" pilot name from my openomf config, booting the game and then loading a character gives me a greyed out new tournament button.

Vagabond commented:
Leaving and re-entering tournament mode seems to fix it.

Vagabond commented:
I set my money to 0 on a fresh character, and lost to cosette twice. I got kicked out of the tournment and was able to rejoin with 0 credits (my credits were negative before). I don't remember what's supposed to happen, though.

mrannanj commented on src/resources/trnmanager.c:
```diff
@@ -9,6 +9,12 @@
+static int trn_sort_compare_fn(const void *a, const void *b) {
+    sd_tournament_file const *trn_a = a;
+    sd_tournament_file const *trn_b = b;
+    return (trn_a->registration_fee > trn_b->registration_fee) - (trn_a->registration_fee < trn_b->registration_fee);
```
What's the problem with just `return trn_a->registration_fee - trn_b->registration_fee;` here?

Nopey replied:
i don't want to overflow or underflow an integer.

Nopey commented:
> I set my money to 0 on a fresh character, and lost to cosette twice. I got kicked out of the tournment and was able to rejoin with 0 credits (my credits were negative before). I don't remember what's supposed to happen, though.

We're missing the "you have to work as a mechanic" and "while working as a mechanic your HAR upgrades / training courses rotted" behaviors.

The credits going negative (after the warning) and then zero (after being kicked out) is correct.

Nopey commented:
pushed more fixes:
- rewrote hardcoded tournament filename behaviors (the original four tournaments are not special-cased, and are not even referenced anywhere in MASTER.DAT).
- `scene` cheat now works with scene names (previously, the strings would never have matching case)
- implemented tournament cutscene replay cheat 'e'

mrannanj commented on src/formats/chr.h:
```diff
@@ -40,7 +40,7 @@ typedef struct {
     float winnings_multiplier;              ///< Tournament winnings multiplier
     sd_chr_enemy *enemies[MAX_CHR_ENEMIES]; ///< List of enemy states in current tournament
-    unsigned int cutscene;                  ///< cutscene id for end of tournament
+    char bk_name[14];                       //< cutscene bk for end of tournament
```
Well, guess use 3 slashes on all since the others have it too

katajakasa replied:
It's doxygen syntax. Reminds me, it has probably been a while since anyone has printed the game docs :/

Vagabond commented on src/formats/chr.h:
```diff
@@ -40,7 +40,8 @@ typedef struct {
     sd_chr_enemy *enemies[MAX_CHR_ENEMIES]; ///< List of enemy states in current tournament
-    unsigned int cutscene;                  ///< cutscene id for end of tournament
+    char bk_name[14];                       ///< cutscene bk for end of tournament
+    int tournament_id;                      ///< misnomer? set to 4 for WORLD.BK cutscene quirks
```
It does correspond to TOURNAMENT_NUMBER in the tournament compiler LST files.

Nopey replied:
Ah, thanks. I did a search for "TOURNAMENT_ID" and came up blank :P
will remove the misnomer comment.

Vagabond commented on src/resources/trnmanager.c:
```diff
@@ -3,26 +3,29 @@
+static int trn_sort_compare_fn(const void *a, const void *b) {
+    sd_tournament_file const *trn_a = a;
+    sd_tournament_file const *trn_b = b;
+    return (trn_a->registration_fee > trn_b->registration_fee) - (trn_a->registration_fee < trn_b->registration_fee);
```
I *believe* the original game sorts by the ID first and then by the fee, if there's more than 1 with the same ID? See https://omf2097.com/wiki/doku.php?id=omf2097:robyrt_tourn

Nopey replied:
thanks, i'll do a quick n' dirty TRN file edit to verify and then use the correct logic.

Nopey replied:
the linked wiki is wrong. tournaments are not sorted by `TOURNAMENT NUMBER`.

Vagabond replied:
Good to know.
